File: montecarlo.rs
struct Coord { x:f64, y:f64 }

fn getrand() -> f64 {
  use std::rand;
  use std::rand::*;
  return rand::random::<f64>();
}

fn in_qsect(coord:Coord) -> bool {
  use std::num::Float;
  return coord.x.powf(2.0) + coord.y.powf(2.0) <= 1.0
}


static ITERATION:u32 = 800_000;

fn main() {
  let mut count_in_qsect:i32 = 0;

  for i in 0..ITERATION {
    if in_qsect(Coord { x:getrand(), y:getrand() }) {
      count_in_qsect += 1;
    }
  }

  println!("In quarter sector = {} / {}", count_in_qsect, ITERATION);
  println!("Estimated Pi: {}", (count_in_qsect as f64) * 4.0 / (ITERATION as f64));
}

File: montecarlo.go
package main

import (
  "fmt"
  "math"
  "math/rand"
  "time"
)

type Coord struct { x, y float64 }

func getrand() float64 {
  return r.Float64()
}

func in_qsect(coord *Coord) bool {
  return math.Pow(coord.x, 2) + math.Pow(coord.y, 2) <= 1
}


const ITERATION = 800000

var r = rand.New(rand.NewSource(time.Now().UnixNano()))

func main() {
  count_in_qsect := 0

  for i := 0; i < ITERATION; i++ {
    if in_qsect( &Coord{ getrand(), getrand() } ) {
      count_in_qsect++
    }
  }

  fmt.Println(`In quarter sector =`, count_in_qsect, `/`, ITERATION)
  fmt.Println(`Estimated Pi:`, float64(count_in_qsect) * 4 / ITERATION)
}
